File: editorial/chap6-16_20/prob6-17.py
# 入力
N = int(input())

# 縦の長さを 1 から √N まで全探索
LIMIT = int(N ** 0.5)
answer = 10 ** 19
for i in range(1, LIMIT + 1):
	if N % i == 0:
		answer = min(answer, 2 * i + 2 * (N // i))

# 答えの出力
print(answer)

File: editorial/chap6-16_20/prob6-17.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 入力
		Scanner sc = new Scanner(System.in);
		long N = sc.nextLong();
		
		// 縦の長さを 1 から √N まで全探索
		long answer = (1L << 60);
		for (long x = 1; x * x <= N; x++) {
			if (N % x == 0) {
				answer = Math.min(answer, 2 * x + 2 * (N / x));
			}
		}
		
		// 答えの出力
		System.out.println(answer);
	}
}